packages/atchops/include/atchops/rsa.h

```diff
@@ -68,16 +69,6 @@ int atchops_rsa_decrypt(const atchops_rsa_key_private_key *private_key, const un
                         const size_t ciphertext_len, unsigned char *plaintext, const size_t plaintext_size,
                         size_t *plaintext_len);
 
-/**
- * @brief generate an RSA keypair
- *
- * @param public_key the public key struct to populate, should be initialized first
- * @param private_key the private key struct to populate, should be initialized first
- * @param keysize the size of the key to generate, e.g. 2048
- */
-int atchops_rsa_generate(atchops_rsa_key_public_key *public_key, atchops_rsa_key_private_key *private_key,
-                         const atchops_md_type md_type);
-
```

Comment:
This was removed from the public API with no replacement.

Reply:
It is redundant code

Reply:
https://github.com/atsign-foundation/at_c/blob/9208717af456f1422d3a92282473b6535562ac84/packages/atchops/include/atchops/rsa_key.h#L70

generation code is in rsa_key.h/.c

Reply:
Have we added support for specifying the size of the key when we generate?

Reply:
Not from an end-developer PoV, I suppose. The way the code is structured allows for 256 or 512-sized keys dynamically and is documented in the code, but if utilised CLI, I don't think this would be known

Reply:
at_c is library only, consumers of at_c may be a CLI but not at_c itself

But, from what I can see in the atchops_rsa_key_generate function, there's no parameter to specify a keysize.

Reply:
Yeah

It seems like the generate function generates a 256 byte modulus (n value) so it only generates a rsa2048 key.

Reply:
Tyler see https://github.com/atsign-foundation/at_c/blob/9208717af456f1422d3a92282473b6535562ac84/packages/atchops/src/rsa_key.c#L244 

Line 244 of rsa_key.c

That should give you a hint as to how to generate a RSA-4096 key

Reply:
Ok, since this PR is for tests, we can do that change in another PR if you want

Reply:
Oh interesting, I wonder what's gone wrong with my test then

Reply:
Well, for generate, it's probably only testing rsa2048. but for hard-code 4096 keys in the tests, everything (in theory) should be working fine

Reply:
Or since the key is hardcoded in the tests at least check against the expected ciphertext